### PR TITLE
Use import in webpack

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- disallows amd require code split points in webpack build
- transpiles `import` to `require` for the karma AMD build

## What is the value of this and can you measure success?

hopefully help #16035

## Tested in CODE?

- [ ] tested

### CODE REVIEW

**Review by gtrufitt — APPROVED** [review]

**Review by SiAdcock — APPROVED** [review]
😍

File: static/src/javascripts/boot-webpack.js [hunk 1]
```diff
@@ -11,6 +11,14 @@ import capturePerfTimings from 'lib/capture-perf-timings';
 // eslint-disable-next-line camelcase,no-undef
 __webpack_public_path__ = `${config.page.assetsPath}javascripts/`;
 
+// Selectively attempt to pollute the global namespace with babel polfills.
+// The polyfills babel provides from core-js will be browser natives where
+// available anyway, so this shouldn't muck about with UAs that don't need it.
+if (!window.Promise) {
+    // used by webpack `import`
+    window.Promise = Promise;
```
Comment (regiskuckaertz) [inline static/src/javascripts/boot-webpack.js]: I don't see the polyfill as a local variable, does it mean this is redundant? i.e. `Promise` is _already_ being overwritten?

File: static/src/javascripts/boot-webpack.js [hunk 2]
```diff
@@ -31,8 +39,7 @@ domready(() => {
         }
 
         userTiming.mark('commercial request');
-        require(
-            ['bootstraps/commercial'],
+        import('bootstraps/commercial').then(
             raven.wrap({ tags: { feature: 'commercial' } }, commercial => {
```
Comment (regiskuckaertz) [inline static/src/javascripts/boot-webpack.js]: Just thinking out loud here. Does `raven.wrap` return whatever the enclosed function expression returns? If yes, then you can reduce to (simplifying):

```
import(...)
.then(raven.wrap(commercial => commercial.init())
.then(() => isEnhanced ? import(...) : () => ())
.then(bootEnhanced => bootEnhanced())
```

File: static/src/javascripts/boot-webpack.js (hunk 1 above)
Comment (sndrs) [inline static/src/javascripts/boot-webpack.js]: babel will provide it when it sees `Promise`

**Comment by sndrs** [comment]
we're going to use `require.ensure`, closing this for now, to reappear later

**Pull request closed without merging**